llvm/lib/IR/RepoTicket.cpp

```diff
@@ -116,7 +116,39 @@ static GODigestState accumulateGOFinalDigestOrInitialDigestAndGetGODigestState(
   return accumulateGOInitialDigestAndGetGODigestState(GO, GOHash, GOIMap);
 }
 
-// Update the GO's hash value by adding the hash of its dependents.
+static bool updateDigestUseContributions(const GlobalObject *GO, MD5 &GOHash,
+                                         const GOVec &Contributions,
+                                         GOInfoMap &GOIMap) {
+  for (const GlobalObject *const G : Contributions) {
+    assert(isa<GlobalVariable>(GO) &&
```

Comment:
This assert could go outside the loop.

Reply:
Thinking about it, maybe this isn't the best place to have these kind of checks.

Is there somewhere where a Contribution is added to a GlobalObject where it could check these? If these assertions were there then it would assert when you tried to create that relationship in the first place.

Or is it fine to create that relationship, but not to calculate a digest of it?

Reply:
Very good point. I have moved the assertion to the code where a Contribution is added to a GOInfoMap and removed the GO parameter. Thanks.